apps/docs/src/examples/extras/instanced-sprite/Trees.svelte

```diff
@@ -0,0 +1,106 @@
+<script lang="ts">
+  import { InstancedSprite, Spritesheet } from '@threlte/extras'
+
+  import { AdaptedPoissonDiscSample as Sampler } from '../../geometry/random-placement/poisson-random/sampling'
+
+  import type { Vector3Tuple } from 'three'
+
+  export let billboarding = false
+
+  const treePositions: Vector3Tuple[] = []
+
+  for (let x = 0; x < 5; x++) {
+    for (let z = 0; z < 5; z++) {
+      treePositions.push([x, 0.5, z])
+    }
+  }
+
+  const REGION_W = 200
+  const REGION_Z = 200
+
+  const greenTrees = [0, 1, 2, 3, 4, 5, 6, 7, 12, 13, 14, 15]
+  const redTrees = [8, 9, 10, 11, 20, 21, 22, 23]
+  const deadTrees = [16, 17, 18, 19]
+
+  const sampler = new Sampler(2, [REGION_W, REGION_Z], undefined, Math.random)
+  const points = sampler.GeneratePoints()
+
+  console.log(points.length)
+
+  const pickRandomTreeType = () => {
+    const rnd = Math.random()
+
+    if (rnd > 0.97) {
+      return `deadTree_${Math.floor(deadTrees.length * Math.random())}`
+    }
+    if (rnd > 0.9) {
+      return `redTree_${Math.floor(redTrees.length * Math.random())}`
+    }
+
+    return `greenTree_${Math.floor(greenTrees.length * Math.random())}`
+  }
+
+  let sprite: any
+
+  $: {
+    // manually update once to apply tree atlas
+    // also, flip random trees on X axis for more variety
+    if (sprite) {
+      for (let i = 0; i < points.length; i++) {
+        sprite.flipX.setAt(i, Math.random() > 0.6 ? true : false)
+      }
+      sprite.update()
+    }
+  }
+</script>
+
+<Spritesheet
```

Comment:
It feels a little strange having configuration as components. Even though it's complex, I would maybe roll all of this up into two components (which would result in one extras export):

```html
<InstancedSprite
  let:Instance
  files={[
    textureUrl: '/textures/sprites/trees-pixelart.png',
    options: { ... },
    animations: [ ... ],
  ]}
>
  <Instance />
  <Instance />
  ...
</InstancedSprite>

Reply:
I see your point and I've reached a similar conclusion today, which is why I tried out a context-builder approach in the `VariantD.svelte` file

I don't like spritesheet configuration being provided as props on `<InstancedSprite/>` because it makes it harder to reuse it. 

Doing something like this would be awkward:

```html
<!-- same behaviour if <InstancedSprite> are wrapped in <Spritesheet> -->
{#await cacodaemonSpritesheet.sheet then { spritesheet, texture }} 
	<!-- hue -->
  <InstancedSprite
    {count}
    {billboarding}
    {spritesheet}
    {texture}
    hueShift={{
      h: 0.3,
      s: 1.5,
      v: 1.5
    }}
  >
    <UpdaterFlyingHook />
  </InstancedSprite>

	<!-- no hue -->
	<InstancedSprite
    {count}
    {billboarding}
    {spritesheet}
    {texture}    
  >
    <UpdaterFlyingHook />
  </InstancedSprite>
	<!-- no hue -->
	<InstancedSprite
    {count}
    {billboarding}
    {spritesheet}
    {texture}    
  >
		<!-- Different behaviour? Team? Culling? -->
    <DifferentUpdater />
  </InstancedSprite>
{/await}
```

https://github.com/threlte/threlte/pull/711/files/8a98a79236c325ca6ad2ca51b847e38991c87cfb#diff-92eff44b60f56ab50d5fa8a0aac2977d897e2bc847f451940ef02df7f1738e58